src/main.cpp

```diff
@@ -2920,6 +2920,16 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         }
     }
 
+    bool pauseLowPrioZendooThread = (
+        fExpensiveChecks &&
+        fScRelatedChecks == flagScRelatedChecks::ON &&
+        fScProofVerification == flagScProofVerification::ON &&
+        SidechainTxsCommitmentBuilder::getEmptyCommitment() != block.hashScTxsCommitment // no sc related tx/certs 
+    );
+       
+    // if necessary pause rust low priority threads in order to speed up times
```

Comment:
Actually we don't need `CZendooLowPrioThreadGuard` at all.
Instead we should create `ProofVerifier` with one more argument `prioritize` same as in mc-cryptolib.
When the prioritized `batchVerified` is triggered, it will automatically pause then resume the low priority threads.